ludwig/schema/hyperopt/executor.py

```diff
@@ -29,6 +29,15 @@ class ExecutorConfig(schema_utils.BaseMarshmallowConfig):
         default=3600, description="The number of seconds for the entire hyperopt run."
     )
 
+    trial_function_resources: Dict[str, float] = schema_utils.Dict(
+        default=None,
+        description=(
+            "The resources reserved by each trial function. This differs from cpu_resources_per_trial and "
```

Comment:
Hm, I think that trial function is close to what is correct, because technically, the trial driver is what calls `tuner.fit`: https://docs.ray.io/en/latest/tune/tutorials/tune-lifecycle.html#what-happens-in-tuner-fit

The trial function resources are for the process between the trial driver and the Ray trainer it spawns. IIUC, there are three processes, each spawning the next:
1. the trial driver, which calls tuner.fit(). Each trial is then spawned in the form of
2. the trial function process, which is `run_experiment` and manages the `ray_queue` for the experiment. Within `run_experiment`, the trainer then spawns
3. `HorovodTrainer` process, which actually executes `ludwig.backend.ray.train_fn`. This process reserves its own placement groups.

This param gives resources to process #2.

Reply:
Just synced with @jeffkinnison and @arnavgarg1 to clarify our shared terminology. Here are the three processes again (each spawning the next):
1. the **tune** driver process, which calls `tuner.fit()`. This name is in line with the [Ray docs](https://docs.ray.io/en/latest/tune/tutorials/tune-lifecycle.html#what-happens-in-tuner-fit).
2. the **trial** driver (formerly called the trial function) process. This process executes `run_experiment`, which creates the Ray Queue.
3. the `trainer` process, which executes `train_fn` in its own placement group.

I'll update the references to `trial_function` to `trial_driver` to make sure this is clear.